### PR TITLE
Retrieving Assets from Azure Blob Storage in editor page

### PR DESCRIPTION
**Pull request by tbarlow12**

### CODE REVIEW

File: src/providers/storage/azureBlobStorage.ts [hunk 1]
```diff
@@ -138,13 +141,41 @@ export class AzureCloudStorageService implements IStorageProvider {
             path = path ? [this.options.containerName, path].join("/") : this.options.containerName;
         }
         const files = await this.listFiles(path);
-        return files.map((blobPath) => AssetService.createAssetFromFilePath(blobPath));
+        const result: IAsset[] = [];
+        for (const key of Object.keys(files.entries)) {
+            const url = this.getUrl(files.entries[key].name);
+            const asset = AssetService.createAssetFromFilePath(url);
+            if (asset.type === AssetType.Image || asset.type === AssetType.Video) {
```
Comment (wbreza) [inline src/providers/storage/azureBlobStorage.ts]: I would consider inverting this to check whether or not it is unknown which is the 3rd option in that enum.

File: src/providers/storage/azureBlobStorage.ts [hunk 2]
```diff
@@ -138,13 +141,41 @@ export class AzureCloudStorageService implements IStorageProvider {
...
+            }
+        }
+        return result;
     }
 
     private getService() {
         return AzureStorageBlob.createBlobService(this.options.connectionString);
     }
 
+    private getUrl(blobName: string) {
```
Comment (wbreza) [inline src/providers/storage/azureBlobStorage.ts]: Is this a static call via the library? Or does it make an HTTP request to figure it out?

File: src/providers/storage/azureBlobStorage.ts [hunk 3]
```diff
@@ -138,13 +141,41 @@ export class AzureCloudStorageService implements IStorageProvider {
...
+            this.getHostName(this.options.connectionString),
+        );
+    }
+
+    private getHostName(connectionString: string): string {
+        const accountName = this.getAccountName(connectionString);
+        return `https://${accountName}.blob.core.windows.net`;
+    }
+
+    private getAccountName(connectionString: string): string {
```
Comment (wbreza) [inline src/providers/storage/azureBlobStorage.ts]: This method can be static

File: src/providers/storage/azureBlobStorage.ts [hunk 4]
```diff
@@ -138,13 +141,41 @@ export class AzureCloudStorageService implements IStorageProvider {
             path = path ? [this.options.containerName, path].join("/") : this.options.containerName;
         }
         const files = await this.listFiles(path);
-        return files.map((blobPath) => AssetService.createAssetFromFilePath(blobPath));
+        const result: IAsset[] = [];
```
Comment (wbreza) [inline src/providers/storage/azureBlobStorage.ts]: Any tests needed to get created or updated for these changes?

File: src/providers/storage/azureBlobStorage.ts [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
 import AzureStorageBlob from "../../vendor/azurestoragejs/azure-storage.blob.js";
-import { IAsset } from "../../models/applicationState";
+import { IAsset, AssetType } from "../../models/applicationState";
 import { AssetService } from "../../services/assetService";
 
 export interface IAzureCloudStorageOptions {
     connectionString: string;
     containerName: string;
     createContainer: boolean;
+    accountName?: string;
```
Comment (wbreza) [inline src/providers/storage/azureBlobStorage.ts]: Doesn't look like you are using these outside of the options object here.

File: src/providers/storage/azureBlobStorage.ts (hunk 1 above)
Comment (tbarlow12) [reply src/providers/storage/azureBlobStorage.ts]: Done

File: src/providers/storage/azureBlobStorage.ts (hunk 2 above)
Comment (tbarlow12) [reply src/providers/storage/azureBlobStorage.ts]: I believe it's a static call

File: src/providers/storage/azureBlobStorage.ts (hunk 4 above)
Comment (tbarlow12) [reply src/providers/storage/azureBlobStorage.ts]: I've been playing around with this, I think I'll add them once the new stuff is in. That way it'll more accurately reflect the new data structure

Comment (tbarlow12) [reply src/providers/storage/azureBlobStorage.ts]: nvm, almost done adding tests now

**Pull request merged**